chapters/equations.tex

```diff
@@ -442,7 +442,7 @@ \subsection{reinit}\label{reinit}
 \end{lstlisting}
 
 The operator reinitializes \lstinline!x! with \lstinline!expr! at an event instant.
-\lstinline!x! is a \lstinline!Real! variable (or an array of \lstinline!Real! variables) that must be selected as a state (resp., states), i.e., \lstinline!reinit! on \lstinline!x! implies \lstinline!stateSelect = StateSelect.always! on \lstinline!x!.
+\lstinline!x! is a component-reference (where any subscripts are evaluable) refering to a \lstinline!Real! variable (or an array of \lstinline!Real! variables) that must be selected as a state (resp., states), i.e., \lstinline!reinit! on \lstinline!x! implies \lstinline!stateSelect = StateSelect.always! on \lstinline!x!.
```

Comment:
Typo:
```suggestion
\lstinline!x! is a component-reference (where any subscripts are evaluable) referring to a \lstinline!Real! variable (or an array of \lstinline!Real! variables) that must be selected as a state (resp., states), i.e., \lstinline!reinit! on \lstinline!x! implies \lstinline!stateSelect = StateSelect.always! on \lstinline!x!.
```

Reply:
I can only find one place where we use the hyphened form _component-reference_ (which should be changed):
```suggestion
\lstinline!x! is a component reference (where any subscripts are evaluable) referring to a \lstinline!Real! variable (or an array of \lstinline!Real! variables) that must be selected as a state (resp., states), i.e., \lstinline!reinit! on \lstinline!x! implies \lstinline!stateSelect = StateSelect.always! on \lstinline!x!.
```

Reply:
Now that we speak of a component-reference I find it somewhat confusing to also speak of variables or arrays of variables in the same sentence.  Could be avoided like this?
```suggestion
\lstinline!x! is a \lstinline!Real!-valued (or \lstinline!Real! array) component reference (where any subscripts are evaluable) that must be selected as a state (resp., states), i.e., \lstinline!reinit! on \lstinline!x! implies \lstinline!stateSelect = StateSelect.always! on \lstinline!x!.
```

Reply:
As noted in #3420 I don't see a reason to remove the dash.
For the other part only I don't want to also mention _values_, since only variables can be states and have stateSelect attributes.

A variant would be "referring to Real variable(s)". 
I did consider "one or more" - but realized that reinit can take an empty array as well, so in that case it would "zero or more Real variables that must be selected as states".